jsapp/js/components/processing/sidebar/sidebarDisplaySettings.tsx

```diff
@@ -22,6 +33,25 @@ export default function SidebarDisplaySettings() {
     store.getDisplays(activeTab)
   );
 
+  function getInitialFields() {
+    if (!props.assetContent?.survey) {
+      return [];
+    }
+
+    const questionsList = getFlatQuestionsList(props.assetContent.survey, 0)
+      .filter((question) => !(question.name === store.currentQuestionName))
+      .map((question) => {
+        // We make an object to show the question label to the user but use the
+        // name internally so it works with duplicate question labels
+        return {name: question.name, label: question.label};
+      });
+
+    return questionsList;
+  }
+
+  const [selectedFields, setSelectedFields] = useState(getInitialFields());
```

Comment:
This would possibly be a useless change for now, but it makes sense to always treat the store as the ultimate source of this information. These `selectedFields` here should only be treated as temporary state for when the modal is being open.

Reply:
See my comment about small bug below (one of the last comments) for actual problem with current ways. It's possible the flow needs to be changed a bit, so that it's loading the hidden/selected fields from the store always when modal opens (and for clarity it might also clean up after closing)

Reply:
Should be fixed now :)